api/capi/include/nntrainer.h

```diff
@@ -215,6 +215,49 @@ int ml_train_model_set_optimizer(ml_train_model_h model,
 int ml_train_model_set_dataset(ml_train_model_h model,
                                ml_train_dataset_h dataset);
 
+/**
+ * @brief Save the model after training
+ * @details Use this function to save the current model.
+ * When calling with @a ML_TRAIN_MODEL_SAVE_LOAD_FLAGS_INFERENCE_PARAMS or
+ * ML_TRAIN_MODEL_SAVE_LOAD_FLAGS_TRAINING_PARAMS, the model must be compiled
+ * with @a ml_train_model_compile beforehand.
```

Comment:
if this function called with INFERENCE_PARAMS, then it will save the model just for inference ( Just weights )?
and if with TRAINING_PARAMS, then it save the model including hyper-params as well as inference?

Reply:
Current thinking is... 

- Some fixed configurations like learning rate, optimizers - `ML_TRAIN_MODEL_SAVE_LOAD_FLAGS_TRAINING_CONFIG` training configuration (for now, only for ccapi)
- Training weights like iterations, adam weights - `ML_TRAIN_MODEL_SAVE_LOAD_FLAGS_TRAINING_PARAMS` (this is exposed)